packages/block-library/src/cover/shared.js

```diff
@@ -156,47 +160,66 @@ function retrieveFastAverageColor() {
  * See the comments below for more details about which aspects take priority when
  * calculating the relative darkness of the Cover.
  *
- * @param {string} url
  * @param {number} dimRatio
  * @param {string} overlayColor
+ * @param {Object} averageBackgroundColor
  * @return {Promise<boolean>} True if cover should be considered to be dark.
  */
-export async function getCoverIsDark( url, dimRatio = 50, overlayColor ) {
+export function getCoverIsDark(
+	dimRatio = 50,
+	overlayColor,
+	averageBackgroundColor = undefined
+) {
 	const overlay = colord( overlayColor )
 		.alpha( dimRatio / 100 )
 		.toRgb();
 
+	if ( averageBackgroundColor ) {
+		const {
+			value: [ r, g, b, a ],
+		} = averageBackgroundColor;
+		// FAC uses 0-255 for alpha, but colord expects 0-1.
+		const media = { r, g, b, a: a / 255 };
+		const composite = compositeSourceOver( overlay, media );
+		return colord( composite ).isDark();
+	}
+
+	// Assume a white background because it isn't easy to get the actual
+	// parent background color.
+	const background = { r: 255, g: 255, b: 255, a: 1 };
+	const composite = compositeSourceOver( overlay, background );
+	return colord( composite ).isDark();
+}
+
+/**
+ * This method evaluates the average color of a background image
+ *
+ * @param {string} url
+ * @return {object|null} Null the color can't be computed, or the color.
+ */
+export async function getAverageMediaColor( url ) {
 	if ( url ) {
 		try {
 			const imgCrossOrigin = applyFilters(
 				'media.crossOrigin',
 				undefined,
 				url
 			);
-			const {
-				value: [ r, g, b, a ],
-			} = await retrieveFastAverageColor().getColorAsync( url, {
+			const color = await retrieveFastAverageColor().getColorAsync( url, {
 				// Previously the default color was white, but that changed
 				// in v6.0.0 so it has to be manually set now.
-				defaultColor: [ 255, 255, 255, 255 ],
+				defaultColor: [ 0, 0, 0, 255 ],
 				// Errors that come up don't reject the promise, so error
 				// logging has to be silenced with this option.
 				silent: process.env.NODE_ENV === 'production',
 				crossOrigin: imgCrossOrigin,
 			} );
 			// FAC uses 0-255 for alpha, but colord expects 0-1.
-			const media = { r, g, b, a: a / 255 };
-			const composite = compositeSourceOver( overlay, media );
-			return colord( composite ).isDark();
+			return color;
```

Comment:
In many cases FAC logs errors and returns the `defaultColor` instead of throwing errors ([example](https://github.com/fast-average-color/fast-average-color/blob/master/src/index.ts#L97-L100)). So those errors return black, but other errors that they don't catch return `null`. I wonder if we should just return the same `defaultColor` on the errors that we catch since we already can't differentiate between some errors and a black image.

Also, since the `defaultColor` switched to black, the behavior here deviates from `getCoverIsDark` which [assumes a white default](https://github.com/WordPress/gutenberg/pull/54054/files#diff-98e8fbed8e10d45fee8dfcb0373d733aa1335f7cbf0a5dd883739b86fae0fd7eR189). I wonder if those should be the same?

Reply:
In `getCoverIsDark` the default is white because we want to see how much it darkens it. I don't think they're related. I can't say we should return black for all errors, it makes little sense. Plus we should then return the full color object FAC builds, but we';'d have to simulate it hence we'll have to always keep it up to date. I'd stick to null.

Reply:
> Plus we should then return the full color object FAC builds, but we';'d have to simulate it hence we'll have to always keep it up to date. 

Or it could return a `Colord` or `RgbaColor` object since we're already using `colord` everywhere else.

Reply:
> In `getCoverIsDark` the default is white because we want to see how much it darkens it. I don't think they're related. 

In one case we're assuming the `overlayColor` goes on white and the other black.

The white default existed because FAC was hardcoded to white at one point. And I think `getCoverIsDark` was mistakenly changed to return `true` by default in #53961—it should have been `false` to match the FAC default.